### PR TITLE
[WIP] ACLK Socks 5 Proxy Support

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
Don't review yet. I will add details before removing WIP.

currently Working SOCKS5 support. Some common code for HTTP proxy support also added but not tested yet.

### CODE REVIEW

File: aclk/aclk_common.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#include "aclk_common.h"
+
+struct {
+    ACLK_PROXY_TYPE type;
+    const char *url_str;
+} supported_proxy_types[] = {
+    { .type = HTTP,     "http://"    },
```
Comment (thiagoftsm) [inline aclk/aclk_common.c]: Should we not add `https://` to the list?

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=5032175&bid=16579332)

File: aclk/aclk_common.c (hunk 1 above)
Comment (underhood) [reply aclk/aclk_common.c]: Hi that will be another PR, i removed http from this PR completely

Comment (thiagoftsm) [reply aclk/aclk_common.c]: All right, thank you!

**Pull request closed without merging**